### PR TITLE
Show nicer error message in ValType-parser for floating point

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

File: test/unittests/parser_test.cpp [hunk 1]
```diff
@@ -45,8 +45,14 @@ TEST(parser, valtype)
     EXPECT_EQ(std::get<0>(parse<ValType>(&b)), ValType::i64);
     b = 0x7f;
     EXPECT_EQ(std::get<0>(parse<ValType>(&b)), ValType::i32);
+    b = 0x7c;
+    EXPECT_THROW_MESSAGE(
+        std::get<0>(parse<ValType>(&b)), parser_error, "unsupported valtype (floating point)");
```
Comment (gumb0) [inline test/unittests/parser_test.cpp]: These don't need `std::get<0>`

Comment (axic) [reply test/unittests/parser_test.cpp]: How would you rewrite them?

Comment (gumb0) [reply test/unittests/parser_test.cpp]: EXPECT_THROW_MESSAGE(parse<ValType>(&b), parser_error, "unsupported valtype (floating point)");

Comment (axic) [reply test/unittests/parser_test.cpp]: Oh right, for the failing case :)

I thought you also meant for the success cases.

Comment (axic) [reply test/unittests/parser_test.cpp]: Fixed.

**Pull request merged**